arcgis-ios-sdk-samples/Layers/Apply unique values with alternate symbols/ApplyUniqueValuesAlternateSymbolsViewController.swift

```diff
@@ -0,0 +1,118 @@
+// Copyright © 2022 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class ApplyUniqueValuesAlternateSymbolsViewController: UIViewController {
+    @IBOutlet var mapView: AGSMapView! {
+        didSet {
+            // Set the map and its initial viewpoint.
+            mapView.map = AGSMap(basemapStyle: .arcGISTopographic)
+            mapView.setViewpoint(AGSViewpoint(center: AGSPoint(x: -13632095.660131, y: 4545009.846004, spatialReference: .webMercator()), scale: 7500))
+        }
+    }
+    
+    @IBOutlet var currentScaleLabel: UILabel!
+    @IBOutlet var resetViewpointBarButtonItem: UIBarButtonItem!
+    
+    /// Response to the bar button item being tapped.
+    @IBAction func resetViewpointTapped(_ button: UIBarButtonItem) {
+        // Create the initial viewpoint.
+        let viewpoint = AGSViewpoint(center: AGSPoint(x: -13631205.660131, y: 4546829.846004, spatialReference: .webMercator()), scale: 7500)
+        // Set the viewpoint with animation.
+        mapView.setViewpoint(viewpoint, duration: 5, curve: AGSAnimationCurve.easeInOutSine) { (finishedWithoutInterruption) in
+            if finishedWithoutInterruption {
```

Comment:
I'm a bit confused by the logic here. If the animation finishes without interruption, shouldn't it just do nothing? Do you mean to check the opposite - if the animation is interrupted, run the button tapped action again?

Either way I think it is a bit overkill here. If the animation is interrupted, just let the user tap the button again. As long as we are not setting the button `enabled` state, no need to deal with animation completion block.